presto-main/src/main/java/com/facebook/presto/server/protocol/QueuedStatementResource.java

```diff
@@ -242,14 +258,20 @@ public void getStatus(
             @Suspended AsyncResponse asyncResponse)
     {
         Query query = getQuery(queryId, slug);
-
+        ListenableFuture<Double> acquirePermitAsync = queryRateLimiter.acquire(queryId);
+        ListenableFuture<?> waitForDispatchedAsync = transformAsync(
+                acquirePermitAsync,
+                acquirePermitTime -> {
```

Comment:
* Let's add the unit--`acquirePermitTimeSeconds`
* Seconds is probably too coarse of a grain, since we expect in the vast majority of cases that there will be no blocking at all.  Let's use nanoseconds.
* Additionally, if we use nanoseconds, we can avoid the new to instantiate a new `Duration`, and instead just directly supply the nanosecond value.